dataprofiler/profilers/profile_builder.py

```diff
@@ -317,12 +320,13 @@ def clean_data_and_get_base_stats(df_series, sample_size,
         len_df = len(df_series)
         if not len_df:
             return df_series, {
-                "sample_size": 0, "null_count": 0,
+                "sample_size": 0, "sample_ids": [], "null_count": 0,
                 "null_types": dict(), "sample": []
             }
 
         # Pandas reads empty values in the csv files as nan
         df_series = df_series.apply(str)
+        original_index = df_series.index
```

Comment:
I don't really like this approach because if this happens outside the function it'll require less memory. Just select the series with the smallest set of sample_ids used, then utilize that

Reply:
The problem is that df_series is mutated within this function to contain only the non-null values, so pulling out its index after the fact won't have the index values which are null, resulting in a potentially empty intersection every time it is calculated in `_update_row_statistics`

Reply:
I believe sample_ids should contain the entry locations and is external to this function, then you can get the index locations external to this function by doing:

`df.index[sample_ids[:threshold]]`